src/main/java/com/st/ims/utility/AppUtility.java

```diff
@@ -3,72 +3,133 @@
+
 	}
-	
+
 	/*
-	 * This method will take invoice amount and commission percentage and will return
+	 * This method will take order amount and commission percentage and will return
 	 * commission amount.
 	 */
-	private static double getCommissionAmount(double invoiceAmount , double commissionPercentage) {
-		return invoiceAmount*(commissionPercentage/100);
+	private static double getCommissionAmount(double invoiceAmount, double commissionPercentage) {
+		return invoiceAmount * (commissionPercentage / 100);
 	}
-	
-	
+
 	public static InvoiceDetails setDefaultValues(InvoiceDetails invoiceDetail) {
 		try {
 			Invoice invoice = invoiceDetail.getInvoice();
 			invoice.setDueDate(getDueDate(invoice.getCreateDate()));
-			invoice.setCommissionAmount(getCommissionAmount(invoice.getInvoiceAmount(),invoice.getCommissionPercentage()));
+			invoice.setCommissionAmount(
+					getCommissionAmount(invoice.getInvoiceAmount(), invoice.getCommissionPercentage()));
 			invoice.setInvoiceNumber(getInvoiceNumber());
 			invoiceDetail.setInvoice(invoice);
 			return invoiceDetail;
 		} catch (NoSuchAlgorithmException e) {
-			e.printStackTrace();
+			logger.error("Exception while configuring dueDate,invoiceNumber and commissionAmount.", e);
 		}
 		return invoiceDetail;
-		
-		
+
+	}
+
+	public static Order setDefaultValues(Order order) {
+		try {
+			order.setDueDate(getDueDate(order.getCreateDate()));
+			order.setCommissionAmount(getCommissionAmount(order.getOrderAmount(), order.getCommissionPercentage()));
+			order.setOrderNumber(getInvoiceNumber());
+			return order;
+		} catch (NoSuchAlgorithmException e) {
+			logger.error("Exception while configuring dueDate,orderNumber and commissionAmount.", e);
+		}
+		return null;
+	}
+	
+	/* This method will return OrderDetails List with products. */
+	public static List<OrderDetails> getOrderDetailsList(Order order, List<OrderProducts> orderProductList){
+		try {
+			List<OrderDetails> orderDetailsList = new ArrayList<>();
+			 Order filledOrder = setDefaultValues(order);
+			if(!orderProductList.isEmpty()) {
+				for(OrderProducts productList:orderProductList){
+					OrderDetails newOrderDetail = new OrderDetails();
+					newOrderDetail.setOrder(filledOrder);
+					newOrderDetail.setProduct(productList.getProduct());
+					newOrderDetail.setProductCount(productList.getProductQuantity());
+					orderDetailsList.add(newOrderDetail);
+				}
+				return orderDetailsList;
+			}else {
+				logger.error("Order doesn,t have any product associated with it.");
+			}
+		} catch (Exception e) {
+			logger.error("Exception while getting OrderDetailsList.",e);
+		}
+		return Collections.emptyList();
+	}
+	
+	/* This method will create new OrderDetails list for update operation. */
+	public static List<OrderDetails> getUpdatedOrderList(Order order,List<OrderProducts> orderProductList){
+		try {
+			List<OrderDetails> orderDetailsList = new ArrayList<>();
```

Comment:
should decleare with null

Reply:
Compiler doesn,t allows to be null at this location.